zkwebui/WEB-INF/src/org/adempiere/webui/apps/form/WTrialBalance.java

```diff
@@ -323,11 +319,20 @@ private void loadTable()
 		
 		if (fReportCube.getValue() !=null)
 			pa_ReportCube_ID = (Integer)fReportCube.getValue();
-		
-		//pa_ReportCube_ID = 1000000;		
-		if (pa_ReportCube_ID == 0 || c_PeriodTo_ID == 0 || m_AD_Org_ID == 0)
-		{
-			FDialog.error(form.getWindowNo(), "Please fill mandatory fields (Cube, Period, Organization).");
+
+		//pa_ReportCube_ID = 1000000;
+
+		MAcctSchema acctSchema = MAcctSchema.getClientAcctSchema(Env.getCtx(), Env.getAD_Client_ID(Env.getCtx()))[0];
+		if (acctSchema == null)
+			FDialog.error(form.getWindowNo(), "@No Acctschema@");
+		MAcctSchemaElement acctElementOrg = acctSchema.getAcctSchemaElement(MAcctSchemaElement.ELEMENTTYPE_Organization);
+		Boolean orgMandatory = acctElementOrg.isBalanced()?true:false;
+
+		if (pa_ReportCube_ID == 0 || c_PeriodTo_ID == 0 || (orgMandatory && m_AD_Org_ID == 0)){

```

Comment:
Hi @SusanneCalderon i see it but i think that is good practice if you use AD message for show error, maybe you can change it line for:
<pre>
		StringBuffer errorMessage = new StringBuffer();
		if (pa_ReportCube_ID == 0) {
			errorMessage.append("@PA_ReportCube_ID@");
		}
		if (c_PeriodTo_ID == 0) {
			if(errorMessage.length() > 0) {
				errorMessage.append(Env.NL);
			}
			errorMessage.append("@C_PeriodTo_ID@");
		}
		if (orgMandatory && m_AD_Org_ID == 0) {
			if(errorMessage.length() > 0) {
				errorMessage.append(Env.NL);
			}
			errorMessage.append("@AD_Org_ID@");
		}
		if(errorMessage.length() > 0) {
			FDialog.error(form.getWindowNo(), "FillMandatory", errorMessage.toString());
			return;
		}
</>

Reply:
Hi @yamelsenih 
You are right, I will change it

Reply:
Hi @yamelsenih ; I included your suggestion